File: backend/core/models/stock.py
from google.appengine.ext import ndb

from core.models import BaseModel


class Stock(BaseModel):
    """
    This class is for defining the Model of the Entity Stock. Its properties and methods.
    """
    # Properties of the ndb Model
    code = ndb.StringProperty(required=True)
    name = ndb.StringProperty()
    price = ndb.FloatProperty()
    pe = ndb.FloatProperty()
    price_sale = ndb.FloatProperty()
    eps = ndb.FloatProperty()
    total_cash_per_share = ndb.FloatProperty()
    book_value_per_share = ndb.FloatProperty()
    dividend_yield = ndb.FloatProperty()
    profit_margin = ndb.FloatProperty()
    total_debt = ndb.FloatProperty()
    notes = ndb.StringProperty()
    shares = ndb.StringProperty()
    shares_num = ndb.FloatProperty()
    years_debt = ndb.FloatProperty()

    @classmethod
    def get_by_code(cls, code):
        """
        This property from Stock is useful for getting the entity by its code string
        """
        complete_code = 'ASX:'+code
        s = cls.query(cls.code == complete_code).get()
        if s:
            return s
        else:
            return None

File: frontend/app/models/stock.js
/*
In models we define the structure of the classes that we use in the Backend
*/
import DS from 'ember-data';  // Import the ember data module

var Stock = DS.Model.extend({ // Define the stock model
  code: DS.attr('string'),  // Code as string
  companyName: DS.attr('string'), // companyName as string
  currentPrice: DS.attr('number'), // currentPrice as string
  pe: DS.attr('number'),
  priceSales: DS.attr('number'),
  eps: DS.attr('number'),
  totalCashPerShare: DS.attr('number'),
  bookValuePerShare: DS.attr('number'),
  dividendYield: DS.attr('string'),
  profitMargin: DS.attr('number'),
  totalDebt: DS.attr('number'),
  notes: DS.attr('string'),
  yearsDebt: DS.attr('string'),
  shares: DS.attr('string')
});

export default Stock; // Export the model